### PR TITLE
WT-6364 Fix double counting sizes of existing updates on the update chain for the history store

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/btree/row_modify.c [hunk 1]
```diff
@@ -113,9 +113,18 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
...
 
+            /* If there are existing updates, append them after the new updates. */
+            if (cbt->compare == 0) {
+                for (last_upd = upd; last_upd->next != NULL; last_upd = last_upd->next)
+                    ;
+                if (cbt->ins != NULL)
+                    last_upd->next = cbt->ins->upd;
+                else if ((mod = cbt->ref->page->modify) != NULL && mod->mod_row_update != NULL)
+                    last_upd->next = mod->mod_row_update[cbt->slot];
+            }
```
Comment (michaelcahill) [inline src/btree/row_modify.c:124]: I think you can replace all of this with:
```c
                for (last_upd = upd; last_upd->next != NULL; last_upd = last_upd->next)
                     ;
                last_upd->next = *upd_entry;
```

Comment (michaelcahill) [reply src/btree/row_modify.c:124]: Also looking at this, I think there is an existing bug if there is an error: we end up calling `__wt_free_update_list` in the calling code, which will free the updates that are already attached to the tree.

In other words, we need to set `last_upd->next = NULL` on error before returning.

Comment (quchenhao) [reply src/btree/row_modify.c:122]: Done

**Comment by quchenhao** [comment]
@tammybailey I added luke as reviewer.

File: src/btree/row_modify.c (hunk 1 above)
Comment (michaelcahill) [reply src/btree/row_modify.c:122]: You don't need to re-check `cbt->compare == 0` either.

**Pull request merged**